jax/_src/api.py

```diff
@@ -675,6 +675,15 @@ def xla_computation(fun: Callable,
 
   fun_name = getattr(fun, "__name__", "unknown")
 
+  if getattr(fun, "_is_pmap", False):
+    # TODO(tomhennigan): Unbox the pmap and set axis_env rather than throwing?
+    raise ValueError(
+        "`xla_computation(pmap(f, 'i'))(x)` is equivalent to adding an outer "
+        "jit to your pmap'd computation making it hard to interpret the "
+        "resulting xla computation. To get the xla_computation of a pmap'd "
+        "function you should instead use: "
+        "`xla_computation(f, axis_env=[('i', x.shape[0])])(x[0])`")
```

Comment:
I'm not sure how I feel about making this an error instead of a warning 😕 @mattjj any preferences?

One other minor issue here is that usually `xla_computation` accepts anything that has `.shape` and `.dtype` as an arg, but it doesn't require it to define `__getitem__`, so the recommendation might not always work.

Reply:
I am also unsure, and could easily be convinced otherwise, but I lean towards warning instead of error because
1. people will see the warning and so be alerted to the possibly undesirable behavior and be able to rewrite the code to get what they want, disarming the footgun;
2. some folks might _want_ to do xla_computation-of-pmap, perhaps in some nested pmap case (there could even be checked-in tests that already do this...).

I'm not sure if that first premise is true. Maybe some peoples' eyeballs automatically filter out warnings?

Do you folks buy those premises, and further that the premises suggest we should warn instead of error? Are there other premises worth considering?